benefits/core/widgets.py

```diff
@@ -15,3 +16,26 @@ def __init__(self, pattern=None, placeholder=None, **kwargs):
             self.attrs.update({"pattern": pattern})
         if placeholder:
             self.attrs.update({"placeholder": placeholder})
+
+
+class RadioSelect(widgets.RadioSelect):
+    template_name = "core/widgets/radio_select.html"
+    option_template_name = "core/widgets/radio_select_option.html"
+
+    def __init__(self, choice_descriptions=(), *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.choice_descriptions = list(choice_descriptions)
+
+    def __deepcopy__(self, memo):
+        obj = super().__deepcopy__(memo)
+        obj.choice_descriptions = copy.copy(self.choice_descriptions)
+        return obj
+
+    def create_option(self, name, value, label, selected, index, subindex, attrs):
+        option = super().create_option(name, value, label, selected, index, subindex, attrs)
+        # this implementation does not support groups from ChoiceWidget.optgroups
+        for choice_value, choice_description in self.choice_descriptions:
```

Comment:
See comments below about using a `dict` for `choice_descriptions` instead of list of tuples - will make this code simpler

Reply:
Thanks for pairing with me on all this through screenshare earlier! Commenting here with some links here that explain more on what these "groups" are and to document that we decided we don't need to support these for now.

https://docs.djangoproject.com/en/3.2/ref/models/fields/#field-choices-named-groups
https://code.djangoproject.com/ticket/20931
https://github.com/django/django/blob/3.2/django/forms/widgets.py#L587
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/optgroup